src/web.ts

```diff
@@ -1,15 +1,150 @@
 import { WebPlugin } from '@capacitor/core';
-
 import type { GoogleAuthPlugin, InitOptions, User } from './definitions';
 
+const GAPI_SCRIPT_ID = 'gapi' as const;
+const GAPI_SRC = 'https://apis.google.com/js/platform.js' as const;
+
+type InitializationErrorKind =
+  | 'NoDocumentOrWindow'
+  | 'DuplicateScript'
+  | 'NoClientId'
+  | 'MissingOptions';
+
+export class InitializationError extends Error {
+  constructor(kind: InitializationErrorKind) {
+    super(
+      `GoogleAuthPlugin error: ${
+        {
+          NoDocumentOrWindow: 'Missing html <document> element',
+          DuplicateScript: `The script tag <script id="${GAPI_SCRIPT_ID}"> already exists`,
+          NoClientId: 'Missing web client ID',
+          MissingOptions:
+            'The unthinkable happened and plugin options are missing',
+        }[kind]
+      }`,
+    );
+  }
+}
+
 export class GoogleAuthWeb extends WebPlugin implements GoogleAuthPlugin {
-  initialize(_options?: InitOptions): Promise<void> {
-    throw this.unimplemented('Initialize is not yet available on the web');
+  options?: InitOptions;
+
+  async initialize(
+    _options: Partial<InitOptions> = {
+      clientId: '',
+      scopes: [],
+    },
+  ): Promise<void> {
+    const metaClientId = (
+      document.getElementsByName(
+        'google-signin-client_id',
+      )[0] as HTMLMetaElement
```

Comment:
This cast should be replaced by an `instanceof` check in my opinion.

Also, the way it is now, `metaClientId`'s type does not include `undefined`, which is incorrect.